### PR TITLE
Frontend/edit post

### PR DESCRIPTION
**Pull request by kittybest**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`yarn build`) was run locally and any changes were pushed
- [x] Lint (`yarn lint --check`) has passed locally and any fixes were made for failures

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

- edit post UI in frontend
-
-

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by vivianjeng** [comment]
1. I think the distance between buttons is not balanced
截圖 2022-11-17 下午3 27 36
2. Etherscan link doesn't dispaly after editing posts
截圖 2022-11-17 下午3 43 55

File: packages/frontend/src/context/Post.ts [hunk 1]
```diff
@@ -307,6 +307,107 @@ export class Data {
...
+                            publicSignals,
+                        }),
+                        method: 'POST',
+                    })
+                    const { transaction, error, post: _post } = await r.json()
+                    if (error) throw error
+                    await queueContext.afterTx(transaction)
+                    const post = this.convertDataToPost(_post)
+                    this.postsById[post.id] = post
+                    this.save()
```
Comment (vivianjeng) [inline packages/frontend/src/context/Post.ts:354]: I think it should return transaction hash here
```ts
return { id: transaction, transactionId: transaction }
```

and what is the `id` used for?

File: packages/frontend/src/context/Post.ts [hunk 2]
```diff
@@ -307,6 +307,107 @@ export class Data {
...
```
Comment (vivianjeng) [inline packages/frontend/src/context/Post.ts:401]: same here

File: packages/frontend/src/context/Post.ts [hunk 3]
```diff
@@ -445,6 +546,115 @@ export class Data {
...
+                        transaction,
+                        error,
+                        comment: _comment,
+                    } = await r.json()
+                    if (error) throw error
+                    await queueContext.afterTx(transaction)
+                    await Promise.all([
+                        this.loadCommentsByPostId(_comment.postId),
+                    ])
+                    this.save()
```
Comment (vivianjeng) [inline packages/frontend/src/context/Post.ts:648]: same here

**Review by vivianjeng — CHANGES_REQUESTED** [review]
I think overall looks good for me
Just need to fix the etherscan link

**Comment by kittybest** [comment]
> 1. I think the distance between buttons is not balanced
> 
> 截圖 2022-11-17 下午3 27 36
> 

yep because the `share` and `edit` won't have number to display, so I let it smaller

File: packages/frontend/src/context/Post.ts (hunk 1 above)
Comment (kittybest) [reply packages/frontend/src/context/Post.ts:354]: where is the `{ id: transaction, transactionId: transaction }`

Comment (vivianjeng) [reply packages/frontend/src/context/Post.ts:354]: https://github.com/Unirep/Unirep-Social/blob/9d8473d9e9a1060d61b76b7c3af0c12bfd4761c2/packages/frontend/src/context/Post.ts#L301

Comment (vivianjeng) [reply packages/frontend/src/context/Post.ts:354]: so I mean the `editPost` function should return  
```ts
    return { id: transaction, transactionId: transaction }
```
as in `publishPost`
but not sure what is the `id` used for in `publishPost`

**Review by AnthonyMadia — COMMENTED** [review]
Two things for consideration:

1. Should we limit the number of times a user can edit a post? Editing a post seems like a feature that can be taken advantage of. 
2. Is there a way to disable the `Update` button if the user has not actually made any edits to their post? I think this would make for better UX.

Otherwise, great work.

**Comment by kittybest** [comment]
@vivianjeng please check if this is what you mean :)

**Comment by kittybest** [comment]
@AnthonyMadia although the style needs to be changed, is it what you mean by disable update while user doesn't make any changes?

File: packages/frontend/src/context/Post.ts [hunk 4]
```diff
@@ -594,6 +598,11 @@ export class Data {
                         this.loadCommentsByPostId(_comment.postId),
                     ])
                     this.save()
+
+                    return {
+                        id: _comment.postId + '#' + transaction,
```
Comment (vivianjeng) [inline packages/frontend/src/context/Post.ts:603]: The url should be postID (nanoID) # commentID (nanoID), is it?
so it should be `id: _comment.postId + '#' + _comment._id `

File: packages/frontend/src/context/Post.ts [hunk 5]
```diff
@@ -646,6 +655,11 @@ export class Data {
...
```
Comment (vivianjeng) [inline packages/frontend/src/context/Post.ts:660]: same here

**Review by vivianjeng — CHANGES_REQUESTED** [review]
yeah it should return the `transactionHash`
but also I mean what is the `id` used for because in progress bar it doesn't link to the post (or  comment)
And if it does, please check if the return `id` is correct and if it works (to link to the post)
https://github.com/Unirep/Unirep-Social/blob/506f8a31956f713bf3ae4f46fd503a6763507351/packages/frontend/src/layout/progressBar/progressBar.tsx#L138

**Comment by kittybest** [comment]
@vivianjeng you're right, i'm just copy and paste and forgot to change it >< thx!

**Comment by AnthonyMadia** [comment]
> @AnthonyMadia although the style needs to be changed, is it what you mean by disable update while user doesn't make any changes?

Yes. If a user wants to edit a post, and they do not make any changes, then they should not be able to still press the update button.

**Comment by vivianjeng** [comment]
@kittybest Sorry I think it should be `id: _comment._id` in this commitment?
[45d6d1f](https://github.com/Unirep/Unirep-Social/pull/123/commits/45d6d1fa4ae4eee799637faab4621040d45607b1)

**Comment by kittybest** [comment]
@vivianjeng edited in the newest commit

**Comment by vivianjeng** [comment]
> @vivianjeng edited in the newest commit

It is still `id: _comment.postId + '#' + _comment.id,`
but it should be `id: _comment.id,`

**Comment by kittybest** [comment]
Ah I forgot to push... sorry, now it might be right

**Pull request merged**